Minor improvements in grammar and clarity (#1831)
usnistgov/OSCAL: src/metaschema/readme.md

OSCAL Metaschemas are used to generate other OSCAL artifacts based on the metaschema description. These artifacts include:

- XML and JSON schemas, or *data models*, which are derived from the metaschema and expressed as XSD and JSON Schema;
- Conversion scripts that can convert OSCAL content between the XML and JSON OSCAL formats, ensuring the resulting content is schema valid;
- XML and JSON [model documentation](https://pages.nist.gov/OSCAL/docs/schemas/) for use on the [OSCAL website](https://pages.nist.gov/OSCAL).

> Use of Metaschemas in OSCAL allows us to maintain seamless and consistent support for multiple OSCAL model formats, including XML and JSON. Content can also be kept up-to-date in multiple formats using generated content converters, and can be validated using generated schemas. Adding support for new formats (e.g., YAML) can accomplished by extending the Metaschema tooling to produce schemas and converters for other formats.

Reviewer: Blocking: I believe Metaschema maintainers now use the terminology Metaschema modules to be less ambiguous and confusing when talking about Metaschema (MIMF, aka the Metaschema Information Modeling Framework with the more formal title) versus "metaschemas" now being Metaschema modules (one or more model documents defined using MIMF), I would think we should align on that. Additionally, we seem to be repeating a lot of information that should be on the Metaschema site, so it makes more sense to point them there I think.

```suggestion
Use of Metaschema modules in OSCAL allows us to maintain seamless and consistent support for multiple OSCAL model formats, including XML and JSON. More information can be found at the [Metaschema Information Modeling Framework documentation site](https://pages.nist.gov/metaschema/)
```

Author: I updated the terminology and included the link, in 11dfbb82a7f00973a0cfd911911a0a2ecc27214e.

> The Metaschema syntax (an XML application) is also described and constrained with its [own schema](https://github.com/usnistgov/metaschema/blob/main/toolchains/xslt-M4/validate/metaschema.xsd), and with a [Schematron constraints set](https://github.com/usnistgov/metaschema/blob/main/toolchains/xslt-M4/validate/metaschema-composition-check.sch). The latter is able to enforce some of the rules described below.

Reviewer: Blocking: can you remove this line **and everything inclusive below it**, I would prefer to remove all of it. (The GitHub ````suggestion` notation will not support the large suggestion to delete it all, I tried multiple times.)

```suggestion
```

Author: I removed all that content, in 11dfbb82a7f00973a0cfd911911a0a2ecc27214e.

Thanks for pointing me to another resource for my learning about Metaschema. :)

OSCAL Metaschemas are used to generate other OSCAL artifacts based on the metaschema description. These artifacts include:

> - XML and JSON schemas, or *data models*, which are derived from the metaschema and expressed as XSD and JSON Schema;

Reviewer: Minor, non-blocking: one last terminology update I missed, but not a big deal.

```suggestion
- XML and JSON schemas, or *data models*, which are derived from the Metaschema modules and expressed as XSD and JSON Schema;
```

## Metaschema modeling

In the case of field and flag objects, the modeling constraints to be imposed by the result schemas (either XSD or JSON Schema) over the data set can be determined on the basis of how they are described. Assembly definitions, however, permit not only flags to be assigned to assemblies of the defined type; additionally, they contain a `model` element for a *model declaration*. This declaration names the subcomponents to be permitted (in documents valid to the target schemas) within any assembly of the type being defined.

> Five elements are used within `model` to define permissible contents of assemblies (elements or objects) being defined. Each of these represents a different object type. Flags are not assigned via `model` but directly in the definition; for the model, we can choose either singles or plurals of named fields or assemblies (i.e., a binary choice between cardinality constraints to be applied). This gives us four choices; additionally, we have the opportunity to use an element `prose`, at most once inside any assembly's model.

Author: The reason I added "at most" was that on first reading of "once inside," I didn't know if the word once was meant in a more colloquial way, as in "once you are inside."

Alternatively, you could end the sentence with something like "... an element `prose` in any assembly's model." and rely on line 44 to indicate the multiplicity.

Reviewer: I think your impressions are very good, but I have a larger piece of feedback that will obsolete this feedback. It is good, but I want to go even leaner. 😄
